entity-service-client/src/main/java/org/hypertrace/entity/data/service/client/EdsCacheClient.java

```diff
@@ -45,12 +54,31 @@ public EdsCacheClient(EntityDataServiceClient client) {
             });
 
     this.entityCache =
-        CacheBuilder.newBuilder().expireAfterWrite(5, TimeUnit.MINUTES).maximumSize(1000)
+        CacheBuilder.newBuilder()
+            .expireAfterWrite(cacheConfig.getEntityCacheExpiryMs(), TimeUnit.MILLISECONDS)
+            .maximumSize(cacheConfig.getEntityMaxCacheSize())
             .build(new CacheLoader<>() {
               public Optional<Entity> load(@Nonnull EdsCacheKey key) {
                 return Optional.ofNullable(client.getById(key.tenantId, key.entityId));
               }
             });
+
+    this.entityIdsCache = CacheBuilder.newBuilder()
+        .expireAfterWrite(cacheConfig.getEntityIdsCacheExpiryMs(), TimeUnit.MILLISECONDS)
+        .maximumSize(cacheConfig.getEntityIdsMaxCacheSize())
+        .build(new CacheLoader<>() {
+          public Optional<String> load(@Nonnull EdsTypeAndIdAttributesCacheKey key) {
+            Entity entity = client.getByTypeAndIdentifyingAttributes(key.tenantId,
+                key.byTypeAndIdentifyingAttributes);
+            String entityId = null;
+            if (entity != null) {
```

Comment:
Do you want to cache null as Optional.absent() or not cache at all?

Reply:
The former. If you don't cache, you wouldn't even know that we queried for that entry right?

Reply:
done